src/Symfony/Component/DomCrawler/Crawler.php

```diff
@@ -593,6 +593,36 @@ public function html()
     }
 
     /**
+     * Evaluates an XPath expression.
+     *
+     * Since an XPath expression might evaluate to either a simple type or a \DOMDoneList,
+     * this method will return either an array of simple types or a new Crawler instance.
+     *
+     * @param string $xpath An XPath expression
+     *
+     * @return array|Crawler An array of evaluation results or a new Crawler instance
```

Comment:
Yeah, I thought twice before defining the return type. I also don't like it but the  [`DOMXPath::evaluate()`](http://php.net/manual/en/domxpath.evaluate.php#refsect1-domxpath.evaluate-returnvalues) returns a simple type value or DOMNodeList. The result depends on the type of xpath query. I figured that the end user knows what it's gonna be since he wrote the query.

Not sure how we could solve it differently (other than defining two methods and verifying results, or forbidding one type of queries).